drivers/storage/portworx/portworx.go

```diff
@@ -456,6 +457,10 @@ func setSecuritySpecDefaults(toUpdate *corev1alpha1.StorageCluster) {
 	if toUpdate.Spec.Security != nil {
 		if toUpdate.Spec.Security.Enabled {
 			if toUpdate.Spec.Security.Auth != nil && (*toUpdate.Spec.Security.Auth != corev1alpha1.AuthSpec{}) {
+				if toUpdate.Spec.Security.Auth.GuestAccess == nil || (*toUpdate.Spec.Security.Auth.GuestAccess == "") {
```

Comment:
Can we have a test for these conditions? May be just under the existing test `TestStorageClusterDefaultsForSecurity` in portworx_test.go

Reply:
Added a test for default guest access values